mdstcpip/Connections.c

```diff
@@ -214,11 +214,11 @@ int DisconnectConnection(int conid){
   if (c) {
     c->state |= CON_DISCONNECT;
     pthread_cond_broadcast(&c->cond);
-    if (c->state & !CON_DISCONNECT) {
+    if (c->state & ~CON_DISCONNECT) {
```

Comment:
should be (!(c->state & CON_DISCONNECT )) instead.

Reply:
You want to enter the block if any flag bit except CON_DISCONNECT is set
this corresponds to c->state & ~CON_DISCONNECT

if you use !(c->state & CON_DISCONNECT ) the result is true if bit CON_DISCONNECT is false,  I don't' think this is the condition you expected.